### ISSUE TITLE
Different way of selecting objects in paper vs inference script

### ISSUE THREAD

**Issue by soyebn** [weiliu89/caffe#420]
The paper lists the following under inference section,
"By using a confidence threshold of 0.01, we can filter out most boxes.We then apply nms with jaccard
overlap of 0.45 per class and keep the top 200 detections per image."

However inference script present at examples/ssd_detect.ipynb prunes detected objects using th=0.6 and doesn't apply NMS. 
I was trying to replicate result (detection_eval) which was reported while training . But my inference script didn't generate that kind of detection, hence the question. Also I noticed training script was following the method mentioned in the paper.

**Comment by weiliu89** [comment]
The `examples/ssd_detect.ipynb` is used to display high confidence detection results. The score filtering and nms are done in `detection_output_layer.cu`.

**Comment by soyebn** [comment]
Thanks for the quick answer.
After spending more time looking at source code, I understand it better now. 
Detection_eval is based on 11 pt VOC2007 mAP hence one needs to consider all boxes starting from 0.0 precision to 1.0. 
Where as, I was comparing my detection accuracy at one fixed value of (th=0.6), that was the reason for the difference.
